### PR TITLE
Integrate webpack into DicomArchive

### PR DESCRIPTION
**Pull request by jacobpenny**
Running `npm install` required!

This is a proof of concept for migrating our Javascript to use [webpack](http://webpack.github.io/docs/). The files listed in `webpack.config.js` can be compiled by running `webpack`. 

Other modules can be converted in a similar manner.

### CODE REVIEW

File: modules/dicom_archive/jsx/entrypoint.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import DicomArchive from './dicom_archive';
+
+/**
```
Comment (alisterdev) [inline modules/dicom_archive/jsx/entrypoint.js]: If you want to have an entry point file for every module, I think it's sensible to name it `index.js` as it seem to be the convention in node modules

File: package.json [hunk 2]
```diff
@@ -7,17 +7,22 @@
...
+    "babel-loader": "6.2.8",
     "babel-preset-es2015": "^6.6.0",
     "babel-preset-react": "^6.5.0",
     "eslint": "^3.1.1",
     "eslint-config-google": "^0.6.0",
     "eslint-plugin-react": "^5.2.2",
     "grunt": "^1.0.1",
     "grunt-babel": "^6.0.0",
-    "load-grunt-tasks": "^3.5.0"
+    "grunt-run": "0.6.0",
```
Comment (alisterdev) [inline package.json]: It seems that this add-on can run any 'bash' command. Does it mean we can get rid of `grunt-babel` and just use run for this?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  entry: {
+    './modules/dicom_archive/js/entrypoint.js': './modules/dicom_archive/jsx/entrypoint.js'
+  },
+  output: {
+      path: './',
+      filename: '[name]'
+  },
+  module: {
+    loaders: [
```
Comment (alisterdev) [inline webpack.config.js]: Since we're getting here, maybe it's worth adding an 'uglifier' too?

**Comment by alisterdev** [comment]
**Without webpack:**
- Two JS requests: 3.4KB/437ms
- ReactJS request: 146 KB/247 ms

**With webpack:**
- Singe JS request: 4.1KB/280ms 
- React already included

File: Gruntfile.js [hunk 4]
```diff
@@ -65,8 +64,14 @@ module.exports = function(grunt) {
           'modules/reliability/js/columnFormatter.js': 'modules/reliability/jsx/columnFormatter.js'
         }
       }
+    },
+
+    run: {
+      webpack: {
+        cmd: 'webpack'
```
Comment (alisterdev) [inline Gruntfile.js]: When running webpack from terminal, everything works, but when running with grunt, I get this error:
```bash
Running "run:webpack" (run) task
Verifying property run.webpack exists in config...OK
File: [no files]
Options: wait, failOnError=false, quite=false, ready=1000, cwd="/var/www/loris", passArgs=[], itterable=false, readyBufferLength=1024
running webpack with args []
>> Error: spawn webpack ENOENT
Warning: non-zero exit code -2 Use --force to continue.

Aborted due to warnings.
```

File: webpack.config.js (hunk 3 above)
Comment (jacobpenny) [inline webpack.config.js]: Definitely. Also, source maps for dev env

**Comment by jacobpenny** [comment]
Unfortunately your fun fact isn't quite right :(

React is listed as an external in the `webpack.config.js` so we are still incurring the same `146 KB/247 ms`. (this also means it isn't coming from `node_modules` atm). Without doing this each module's bundle will have to include it, bringing the size to ~200kb. The `CommonsChunkPlugin` webpack plugin can help with this, but IMHO we should leave React as an external until we can fully remove it.

File: Gruntfile.js (hunk 4 above)
Comment (jacobpenny) [inline Gruntfile.js]: Are you using a global `grunt`? Not sure that would cause this, but try `./node_modules/.bin/grunt run:webpack`

**Comment by alisterdev** [comment]
@jacobpenny True, didn't notice that.. What is the point of doing `import React from 'react';` then? Is it just for the future where we wouldn't need depend on global react instance?

File: Gruntfile.js (hunk 4 above)
Comment (alisterdev) [inline Gruntfile.js]: @jacobpenny I was using local grunt, but running your command results in the same error.

**Comment by jacobpenny** [comment]
@alisterdev Good point, it should be removed I think. We can add it back when we're ready.

**Comment by alisterdev** [comment]
Everything to works smoothly now!

**Comment by jacobpenny** [comment]
I'd prefer to merge https://github.com/aces/Loris/pull/2522 first and then update this accordingly.

**Comment by driusan** [comment]
There's another pull request that @jacobpenny sent which conflicts with this, he says to close it.

**Pull request closed without merging**